src/ja/user/cli.py

```diff
@@ -1,19 +1,113 @@
 
     def get_server_command(self) -> ServerCommand:
         """!
         Transforms command line arguments into a UserConfig object. Creates and returns a ServerCommand object from the
         command line arguments combined with the general config loaded in the constructor.
         @return: The ServerCommand created from the combined input.
         """
+
+        with open(self._config_path, 'r') as stream:
+            data_loaded = yaml.safe_load(stream)
+
+        #Combining arguments from the user config file with the default arguments.
+        priority:str = "medium" if "priority" not in data_loaded else data_loaded["priority"]
+        threads:int = -1 if "threads" not in data_loaded else data_loaded["threads"]
+        ram: int = 1 if "ram" not in data_loaded else data_loaded["ram"]
+        mount: List[str] = [] if "mount" not in  data_loaded else data_loaded["mount"]
+        email: str = None if "email" not in data_loaded else data_loaded["email"]
+        preemptible: bool = False if "preemtible" not in data_loaded else data_loaded["preemtible"]
+
+        parser: ArgumentParser = ArgumentParser("Started JobAdder: you can add, cancel or a query a job.")
+        #shared options
+        parser.add_argument("--verbosity", "-v", type=int, choices=[0, 1, 2], default=1)
+
+        subparsers = parser.add_subparsers(dest="command")
+        #parser for add command
+        parser_add = subparsers.add_parser("add")
+        parser_add.add_argument("--name", "-n", default=str(os.getuid()) + "get job id", help="Name of the job.")#TODO: get job id
+        parser_add.add_argument("--source", "-s", required=True, help="Path of the source file containing the job.")
+        parser_add.add_argument("--priority", "-p", choices="low medium high urgent 0 1 2 3".split(), default=priority)
+        parser_add.add_argument("--threads", "-t", type=int, default=threads, help="Assign a number of cpu threads required for your job.")
+        parser_add.add_argument("--ram", "--memory", "-r", type=int, default=ram, help="Assign a specific amount of ram to your job")
+        parser_add.add_argument("--preemptible", "--pe", action="store_true", default=preemptible)
+        parser_add.add_argument("--mount", "-m", nargs="+", type=str, default=mount)
+        parser_add.add_argument("--email", "-e", required=email is None, default=email)
+        
+        #parser for query command
+        parser_query = subparsers.add_parser("query")
+        parser_query.add_argument("--name", "--uid", "-n", dest="name", required=True, )
+            #not sure what to put here yet
+
+        #parser for cancel command
+        parser_cancel = subparsers.add_parser("cancel")
+        parser_cancel.add_argument("--name", "-n", required=True, help="the name of the job that needs to be cancelled")
+
+        args: Namespace = parser.parse_args()
+
+        if args.command == "add":
+            ssh: SSHConfig = SSHConfig(os.getuid())
```

Comment:
Calling os.getuid() on the user client is pointless.
What matters is the user id on the server.
Also, SSHConfig doesn't even have uid as a property, right?

Reply:
SSHConfig has username and hostname as mandatory parameters in the constructor, I thought username refers to the local user id.
How should I initiate a SSHConfig instance?

Reply:
username is the name of the user as a string that you would use for login via ssh.
uid is an int that is used on Linux systems as an identifier for users.
The parameters for SSHConfig should be read from command line arguments and/or the default config.

Reply:
I'm currently implementing SSHConnection and I will soon push a branch that implements SSHConfig.
You could rebase your PR onto that branch to save yourself the work of implementing SSHConfig.